zilliqa/src/consensus.rs

```diff
@@ -482,7 +513,24 @@ impl Consensus {
             self.secret_key.node_public_key(),
         );
 
-        Ok(Some((leader, ExternalMessage::NewView(Box::new(new_view)))))
+        Ok(Some((
+            Some(leader),
+            ExternalMessage::NewView(Box::new(new_view)),
+        )))
+    }
+
+    fn get_consensus_timeout_params(&self) -> (u64, u64) {
+        let head_block = self.head_block();
+        let head_block_view = head_block.view();
+        let consensus_timeout_ms = self.config.consensus.consensus_timeout.as_millis() as u64;
+        let time_since_last_view_change = SystemTime::now()
+            .duration_since(self.view.last_timeout())
+            .expect("last timeout seems to be in the future...")
+            .as_millis() as u64;
+        let view_difference = self.view.get_view().saturating_sub(head_block_view);
+        let exponential_backoff_timeout = consensus_timeout_ms * 2u64.pow(view_difference as u32);
```

Comment:
We must calculate the view difference from the high_qc, not the head block: `let view_difference = self.view.get_view().saturating_sub(self.high_qc.view);`